test/hamiltonian_consistency.jl

```diff
@@ -7,6 +7,27 @@ include("testcases.jl")
 using Random
 Random.seed!(0)
 
+function test_matrix_repr_opererator(block, ψ, Hψ; atol=1e-8)
+    known_operators = Union{DFTK.RealSpaceMultiplication,
+                            DFTK.DivAgradOperator,
+                            DFTK.MagneticFieldOperator,
+                            DFTK.NoopOperator,
+                            DFTK.NonlocalOperator,
+                            DFTK.FourierMultiplication}
+    for operator in block.operators
+        @assert operator isa known_operators
+        operator_matrix = nothing
+        try
+            operator_matrix = Matrix(block)
+        catch
+            @info "Matrix of operator $(typeof(operator)) not yet supported"
+        end
+        if operator_matrix ≠ nothing
+            @test norm(operator_matrix * ψ - Hψ) < atol
+        end
```

Comment:
Why not put into try ... catch?

Reply:
I have put `δψkδHψk` also, but it's a bit a mouthful.